lib/ansible/modules/cloud/amazon/api_gateway.py

```diff
@@ -0,0 +1,161 @@
+    required: true
+  state:
+    description:
+      - NOT IMPLEMENTED Create or delete Lambda function
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  api_file:
+    description:
+      - JSON or YAML file containing swagger definitions for API
+    required: true
+  stage:
+    description:
+      - stage API should be deployed to
+    required: false
+author:
+    - 'Michael De La Rue (@mikedlr)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Update API resources for development
+tasks:
+- name: update API
+  lambda:
+    api_id: 'abc123321cba'
+    state: present
+    api_file: my_api.yml
+
+# update definitions and deploy API to production
+tasks:
+- name: deploy API
+  lambda:
+    api_id: 'abc123321cba'
+    state: present
+    api_file: my_api.yml
+    stage: production
+    deploy_desc: Make auth fix available.
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+# Import from Python standard library
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        api_id=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        api_file=dict(type='str', default=None, aliases=['src']),
+        stage=dict(type='str', default=None),
+        deploy_desc=dict(type='str', default=None),
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False,  ) #TODO !!!`
+
+    api_id = module.params.get('api_id')
+    state = module.params.get('state').lower()
+    api_file = module.params.get('api_file')
+    stage= module.params.get('stage')
+    deploy_desc= module.params.get('deploy_desc')
+
+#    check_mode = module.check_mode
+    changed = False
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='Python module "botocore" is missing, please install it')
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install it')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    try:
+        client = boto3_conn(module, conn_type='client', resource='apigateway',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+        module.fail_json(msg=str(e))
+
+    with open(api_file) as f:
+            apidata=f.read()
```

Comment:
Make sure to catch this exception and give a readable error if the API file doesn't exist.